### PR TITLE
Invite people: contacts page

### PR DESCRIPTION
**Pull request by jakob223**
image
cc @keybase/y2ksquad

### CODE REVIEW

**Comment by buoyad** [comment]
First two commits should not be part of this PR right?

**Comment by jakob223** [comment]
rebased them away @buoyad , thanks

File: shared/people/invite-friends/invite-contacts.native.tsx [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+  } else if ('0123456789'.includes(firstLetter)) {
+    return '0-9'
+  } else {
+    return 'Other'
+  }
+}
+const filterAndSectionContacts = memoize((contacts: Contact[], search: string): Section[] => {
+  const sectionMap: Map<string, Contact[]> = new Map()
+  contacts
+    .filter(contact => contact.name.toLowerCase().includes(search) || contact.value.includes(search))
```
Comment (buoyad) [inline shared/people/invite-friends/invite-contacts.native.tsx]: lowercase search above

File: shared/people/invite-friends/invite-contacts.native.tsx [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+      })
+    }
+  }
+  return sections
+})
+
+const InviteContacts = () => {
+  const {contacts, errorMessage, loading} = useContacts()
+  const [search, setSearch] = React.useState('')
+  const [selectedContacts, setSelectedContacts] = React.useState(new Set<string>())
```
Comment (buoyad) [inline shared/people/invite-friends/invite-contacts.native.tsx]: want to differentiate emails and phone numbers in here?

File: shared/people/invite-friends/invite-contacts.native.tsx [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+      <Kb.ListItem2
+        type="Small"
+        firstItem={index === 0}
+        body={
+          <Kb.Box2 direction="vertical" alignItems="flex-start">
+            <Kb.Text type="BodySemibold">{topText}</Kb.Text>
+            {bottomText && <Kb.Text type="BodySmall">{bottomText}</Kb.Text>}
+          </Kb.Box2>
+        }
+        action={<Kb.CheckCircle checked={checked} onCheck={onCheck} style={styles.checkCircle} />}
```
Comment (buoyad) [inline shared/people/invite-friends/invite-contacts.native.tsx]: Let's make the whole row tappable

File: shared/people/invite-friends/invite-contacts.native.tsx [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+          item.pictureUri ? (
+            <Kb.NativeImage style={styles.thumbnail} source={{uri: item.pictureUri}} />
+          ) : (
+            undefined
+          )
+        }
+      />
+    )
+  }
+  const keyExtractor = (item: Contact) => item.value
```
Comment (buoyad) [inline shared/people/invite-friends/invite-contacts.native.tsx]: This is fragile, can we use something more likely to be unique? IIRC native contacts plumb over some kind of ID, maybe that can be used?

File: shared/people/invite-friends/invite-contacts.native.tsx (hunk 2 above)
Comment (jakob223) [reply shared/people/invite-friends/invite-contacts.native.tsx]: I'd rather differentiate on submit.

File: shared/people/invite-friends/invite-contacts.native.tsx [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+            undefined
+          )
+        }
+      />
+    )
+  }
+  const keyExtractor = (item: Contact) => item.id
+  return (
+    <Kb.Box2 direction="vertical" fullWidth={true} fullHeight={true}>
+      <Kb.ModalHeader
```
Comment (buoyad) [inline shared/people/invite-friends/invite-contacts.native.tsx]: Oh, sry  missed this in the first pass. Any reason not to just use `Kb.Modal`

File: shared/people/invite-friends/invite-contacts.native.tsx [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+        leftButton={
+          <Kb.Text type="BodyPrimaryLink" onClick={navUp}>
+            Cancel
+          </Kb.Text>
+        }
+        rightButton={
+          <Kb.Text
+            type="BodyPrimaryLink"
```
Comment (buoyad) [inline shared/people/invite-friends/invite-contacts.native.tsx]: These are `BodyBigLink`

File: shared/people/invite-friends/invite-contacts.native.tsx [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+            onClick={canContinue ? onSubmit : undefined}
+            style={canContinue ? undefined : styles.disabledLink}
+          >
+            Invite
+          </Kb.Text>
+        }
+        title="Invite friends"
```
Comment (buoyad) [inline shared/people/invite-friends/invite-contacts.native.tsx]: and `hideBorder`

File: shared/people/invite-friends/invite-contacts.native.tsx (hunk 2 above)
Comment (jakob223) [reply shared/people/invite-friends/invite-contacts.native.tsx]: heh actually just tried doing the submit code for this and you're totally right. WIll do it in the other PR though unless you have more changes for me here.

Comment (jakob223) [reply shared/people/invite-friends/invite-contacts.native.tsx]: oh, we raced. Will do here with other changes.

File: shared/people/invite-friends/invite-contacts.native.tsx [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+    return '0-9'
+  } else {
+    return 'Other'
+  }
+}
+const filterAndSectionContacts = memoize((contacts: Contact[], search: string): Section[] => {
+  const searchL = search.toLowerCase()
+  const sectionMap: Map<string, Contact[]> = new Map()
+  contacts
+    .filter(contact => contact.name.toLowerCase().includes(searchL) || contact.value.includes(search))
```
Comment (buoyad) [inline shared/people/invite-friends/invite-contacts.native.tsx]: `contact.value.includes(search)` lowercase this one too?

**Pull request merged**